core/trino-main/src/main/java/io/trino/dispatcher/QueuedStatementResource.java

```diff
@@ -121,8 +121,8 @@ public QueuedStatementResource(
         this.dispatchManager = requireNonNull(dispatchManager, "dispatchManager is null");
 
         requireNonNull(dispatchManager, "dispatchManager is null");
-        this.responseExecutor = requireNonNull(executor, "responseExecutor is null").getExecutor();
-        this.timeoutExecutor = requireNonNull(executor, "timeoutExecutor is null").getScheduledExecutor();
+        this.responseExecutor = requireNonNull(executor, "executor is null").getExecutor();
+        this.timeoutExecutor = requireNonNull(executor, "executor is null").getScheduledExecutor();
```

Comment:
nit: Not sure about this change - after all there is a single executor (today).

Reply:
rnn here is about parameter validation